### DISCUSSION TITLE
Extension activation failed

### DISCUSSION THREAD

**Discussion by TiemenBouma** [community/community#68736] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

In VS Code I cannot activate github Copilot. 

Error code:
[ERROR] [default] [2023-09-29T09:18:07.907Z] Extension activation failed: "connect ECONNREFUSED 140.82.121.6:443"

In the troubleshooting this is mensioned, but I dont know how to do that: "To request another token from api.github.com, try signing in and out of GitHub Copilot from your IDE. Once you've logged out, GitHub Copilot will prompt you to sign back in."

I already swiched off my firewall

**Comment by Akash1134** [comment, marked as answer]
Hi @TiemenBouma ,

This error is generally caused by range of network issues.

If you know you are connecting via a proxy, make sure the proxy is configured correctly in your environment. For more information, see "[Configuring network settings for GitHub Copilot](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-network-settings-for-github-copilot#configuring-proxy-settings-for-github-copilot)."

Copilot uses custom code to connect to proxies. This means a proxy setup that is supported by your editor is not necessarily supported by GitHub Copilot. Some common causes for errors related to proxies are:

- **If your proxy's URL starts https://, it is not currently supported by GitHub Copilot.**

- You may need to authenticate to the proxy. GitHub Copilot supports basic authentication or authentication with Kerberos. If you are using Kerberos, ensure you have a valid ticket for the proxy service and that you are using the correct service principal name for the service. For more information, see "[Configuring network settings for GitHub Copilot](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-network-settings-for-github-copilot#authentication-with-kerberos)."

- Note that GitHub Copilot may reject custom certificates. For more information, see "[Troubleshooting certificate-related errors](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot#troubleshooting-certificate-related-errors)."

Thanks!
